app/controlplane/internal/usercontext/apitoken_middleware.go

```diff
@@ -186,7 +186,15 @@ func setCurrentOrgAndAPIToken(ctx context.Context, apiTokenUC *biz.APITokenUseCa
 
 	// Set the current organization and API-Token in the context
 	ctx = entities.WithCurrentOrg(ctx, &entities.Org{Name: org.Name, ID: org.ID, CreatedAt: org.CreatedAt})
-	ctx = entities.WithCurrentAPIToken(ctx, &entities.APIToken{ID: token.ID.String(), CreatedAt: token.CreatedAt, Token: token.JWT})
+
+	ctx = entities.WithCurrentAPIToken(ctx, &entities.APIToken{
+		ID:          token.ID.String(),
+		Name:        token.Name,
+		CreatedAt:   token.CreatedAt,
+		Token:       token.JWT,
+		ProjectID:   token.ProjectID,
```

Comment:
If the token has a ProjectID, is this change checking that the ProjectID in the database is the same?

Reply:
mm, it is not, let me add it

Reply:
Good catch, I was using the projectID from the DB when in reality we need to use the one from the token. 

Now I check both